pkg/armhelpers/graph.go

```diff
@@ -90,12 +97,22 @@ func (az *AzureClient) CreateApp(ctx context.Context, appName, appURL string, re
 	}
 	servicePrincipalResp, err := az.servicePrincipalsClient.Create(ctx, servicePrincipalReq)
 	if err != nil {
-		return "", "", "", err
+		return "", "", "", "", err
 	}
 
 	servicePrincipalObjectID = to.String(servicePrincipalResp.ObjectID)
 
-	return applicationID, servicePrincipalObjectID, servicePrincipalClientSecret, nil
+	return applicationObjectID, applicationID, servicePrincipalObjectID, servicePrincipalClientSecret, nil
+}
+
+// DeleteApp is a simpler method for deleting an application and the associated spn
+func (az *AzureClient) DeleteApp(ctx context.Context, appName, applicationObjectID string) (autorest.Response, error) {
```

Comment:
Although it's verbose, I think `applicationName` would be a more consistent parameter name.